subsys/bluetooth/host/audio/mpl.c

```diff
@@ -2356,10 +2488,28 @@ uint64_t current_group_id_get(void)
 
 void current_group_id_set(uint64_t id)
 {
-	/* TODO: Actually change the group to the given group */
+	struct mpl_group_t *group;
+	bool track_change;
+
 	BT_DBG_OBJ_ID("Group ID to set: ", id);
-	pl.group->id = id;
-	media_proxy_pl_current_group_id_cb(pl.group->id);
+
+	find_group_by_id(&pl, id, &group);
+
+	if (group != NULL) {
+		if (pl.group != group) {
+			/* Change to found group */
+			pl.group = group;
+			do_group_change_notifications(&pl);
+
+			/* And change to first track in group */
+			track_change = do_first_track(&pl);
+			if (track_change) {
+				do_track_change_notifications(&pl);
+			}
+		}
+	}
+
+	BT_DBG("Group not found");
```

Comment:
This looks like it is also always logged

Reply:
Added missing return